tools/gcs-usage-recommender/README.md

```diff
@@ -0,0 +1,247 @@
+the GCP console.
+
+To do this, go to the [console](https://pantheon.corp.google.com/iam-admin/audit?_ga=2.160269630.-2040617453.1540660549).
+![UI](images/audit_log_UI.png)
+Filter on Cloud Storage, and select ADMIN_READ, DATA_READ, AND DATA_WRITE. Then click save.
+
+Now, we will export the logs from Stackdriver to BigQuery.
+
+````bash
+gcloud logging sinks create gcs_usage \
+  bigquery.googleapis.com/projects/${AUDIT_LOG_PROJECT_ID}/datasets/${AUDIT_LOG_DATASET_ID} \
+  --log-filter='resource.type="gcs_bucket" AND
+  (protoPayload.methodName:"storage.buckets.delete" OR
+   protoPayload.methodName:"storage.buckets.create" OR
+   protoPayload.methodName:"storage.buckets.get" OR
+  protoPayload.methodName:"storage.objects.get" OR
+  protoPayload.methodName:"storage.objects.delete" OR
+  protoPayload.methodName:"storage.objects.create")' \
+  --organization=${ORG_ID} --include-children
+````
+
+This command will create and return a service account ID, such as 
+serviceAccount:o125240632470-886280@gcp-sa-logging.iam.gserviceaccount.com.
+
+Note down the account name, as we'll use it in the next step!
+
+<h4> 2.3.3 Allow the service account to write to our dataset </h4> 
+Taking the service account created in the last step, let’s assign to it the 
+narrowest privileges needed for this tool - in this case, “BigQuery Data Editor.”
+
+This can be done a number of ways, and here we’ll just follow the 
+[manual process](https://cloud.google.com/bigquery/docs/dataset-access-controls#controlling_access_to_a_dataset).
+
+<h4> 2.3.4 Test that the dataset is receiving audit log entries </h4>
+
+Create a GCS object in any project in your GCP organization, and you should see 
+an entry arrive in the `${OUTPUT_DATASET_ID}` dataset.
+
+The daily BigQuery tables named `cloudaudit_googleapis_com_activity_<date>` and 
+`cloudaudit_googleapis_com_data_access_<date>` will appear in your project the 
+first time a GCS object/bucket is created or read after the audit log sink is 
+created. If you are not seeing tables being created, you may have a permissions 
+issue. Troubleshoot by looking at [your project activity](https://console.cloud.google.com/home/activity) 
+in the project where you created the GCS resource. There you should see your 
+audit log sink service account creating the daily audit log BigQuery tables.
+
+<h3> 2.4 Scanning for existing GCS objects </h3>
+Currently, you are now collecting the raw audit log events representing GCS 
+objects/buckets. The missing link, however, is the events for the set of 
+buckets/objects which were created before the sink started generating these events.
+
+We’ll run a process to populate a new, separate table in our dataset with an 
+inventory of the currently existing buckets in your GCP organization to fill in 
+this missing link.
+
+This process only needs to be run once and would benefit from a low-latency 
+network location to GCP. In our example we will run it in the cloud shell.

```

Comment:
Cloud shell limited resources might not be the best place for listing this many buckets.

Reply:
Yes, although since it's a one-off run, I would argue that the latency may be tolerated this time.